src/cache/cache_las.c

```diff
@@ -699,13 +770,36 @@ __wt_las_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MU
                 continue;
             }
 
+            /*
+             * If we encounter a modify with preceding updates that have the same timestamp and
+             * transaction id, they need to be squashed into a single lookaside entry to avoid
+             * displacing each other.
+             */
+            if (upd->type == WT_UPDATE_MODIFY && upd->next != NULL &&
+              upd->start_ts == upd->next->start_ts && upd->txnid == upd->next->txnid) {
+                tmp_upd = upd;
+                WT_ERR(__las_squash_modifies(session, cursor, &upd));
+                if (tmp_upd->prepare_state == WT_PREPARE_INPROGRESS)
+                    ++prepared_insert_cnt;
+                ++insert_cnt;
+                continue;
+            }
+
             cursor->set_value(cursor, upd->durable_ts, upd->prepare_state, upd->type, &las_value);
 
             /* Using insert so we don't keep the page pinned longer than necessary. */
             WT_ERR(cursor->insert(cursor));
             ++insert_cnt;
             if (upd->prepare_state == WT_PREPARE_INPROGRESS)
                 ++prepared_insert_cnt;
+
+            /*
+             * If we encounter subsequent updates with the same timestamp, skip them to avoid
+             * clobbering the entry that we just wrote to lookaside. Only the last thing in the
+             * update list for that timestamp matters anyway.
+             */
+            while (upd->next != NULL && upd->start_ts == upd->next->start_ts)
```

Comment:
I realized when working on this change that a variant of this problem can occur with regular updates. This problem is much easier to deal with since we can just ignore subsequent updates instead of squashing them.

Reply:
I agree that skipping them is good. Don't we need to consider the transaction id's also as part of the check? Otherwise only once LAS record is inserted with one transaction id, the may not be visible to some ongoing transactions?

Reply:
I agree with @kommiharibabu, I think transaction ID is relevant.

Reply:
Since these updates are remaining in memory, the lookaside content only matters if we evict and then subsequently read the page back into the cache right? In that case, I don't see how it's possible to have an ongoing transaction from before the eviction.

Reply:
I'm not following, why can't transactional readers span page eviction?

But, more simply, what if we're running w/o timestamps, this will discard all updates because the timestamps will all be `WT_TS_NONE`, won't they?

Reply:
6253dde